refactor map distance display

moving-bits opened the pull request:
While working on #9282 I've stumbled upon the old, manual implementation of drawing distance infos in `CGeoMap`. Refactored this into using a couple of `TextView` elements, and merged layouts and drawing handling for all maps into one.

![image](https://user-images.githubusercontent.com/3754370/97790899-1f29a280-1bcd-11eb-8449-200b36c39055.png)

This is a prerequisite for #9282.

Lineflyer commented:
Two questions for this:
1. I have to admit I do not understand what is changed here (reading your explanation....cannot read the code :( ). Any visual or functional change or "just" refactoring?
2. This made me think about another observation (slightly offtopic): While updating the manual I have seen, that on Google maps the individual route length is no shown on Google Maps (See screenshot on https://manual.cgeo.org/de/brouter#individuelle_route). Is this a bug or a missing setting? (Feel free to open a dedicated issue in case of bug).

moving-bits commented:
> 1. I have to admit I do not understand what is changed here (reading your explanation....cannot read the code :( ). Any visual or functional change or "just" refactoring?

So far it's just refactoring. `CGeoMap` did not make use of `TextView` overlays put on top of the map, but actually painted the distances onto the map's surface (for reasons unknown to me). So I changed this to using the same mechanism `NewMap` uses and, along the way, extracted the common code parts into separate files, reducing code duplication.

Changing it to a `TextView` is also a prerequisite for #9282, as only when having the display in a separate GUI item (`TextView` in this case) I can attach an `onClick` handler to it for switching settings.

> 2. This made me think about another observation (slightly offtopic): While updating the manual I have seen, that on Google maps the individual route length is no shown on Google Maps (See screenshot on [manual.cgeo.org/de/brouter#individuelle_route](https://manual.cgeo.org/de/brouter#individuelle_route)). Is this a bug or a missing setting? (Feel free to open a dedicated issue in case of bug).

For me it's shown here with 2020.10.29, just not updated on removal of an item - that's #9256, already fixed on `master`. (AFAICS the individual route's length is shown, but as you have no navigation target set, there's still only one distance being displayed.)
